### DISCUSSION TITLE
CVRP Modelling

### DISCUSSION THREAD

**Discussion by koftezz** [google/or-tools#2471] [CP-SAT questions] [answered]
Hi everyone,

I'm trying to implement the CVRP model in Python as can be seen below:
![image](https://user-images.githubusercontent.com/45968481/113051953-405be280-91af-11eb-85f5-76094c9c0997.png)
 
I know there is a routing module for this but I need to implement it like this. However there is a problem on writing a constraint with `>>` in or-tools, I saw a channeling constraint section under CP-SAT solver and mentions a method called `OnlyEnforceIf`. However, I can't define a continuous variable inside CP-SAT solver. I can define it in gurobi easily. Can someone help?

**Comment by lperron** [comment]
why do you need continuous variables for a cvrp problem ?

**Reply by koftezz** [reply]
Because in my case, I need an continuous variable to track the cumulative demand., which is `u(i)`

**Comment by lperron** [comment, marked as answer]
Using continuous variables is a modeling trick used in the MIP community to
speed up solving by telling the solver not to branch on these variables.

Logically, you can use integer variables, provided you scale all distances
to integral value.
Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00

**Reply by lperron** [reply]
And the solver offers the Routes and the Circuit constraints, which would be much better for modelling TSP and (C)VRP problems.
